### ISSUE TITLE
[1.3-modules-beta.2] Ensure smooth upgrade from Meteor 1.2

### ISSUE THREAD

**Issue by trusktr** [meteor/meteor#5884]
I think the meaning of "lazy" in the [docs](https://github.com/meteor/meteor/blob/release-1.3/packages/modules/README.md#modular-application-structure) means files in an `imports` folder won't be loaded unless they are imported by another file.

I've got `client/imports/foo/index.js`

```js
import $ from 'jquery'
console.log($)
```

which isn't imported by anything. Chrome says

```
Uncaught SyntaxError: Unexpected token import
```

and the Sources tab shows

```
(function(){
import $ from 'jquery'
console.log($)

}).call(this);
```

for `client/imports/foo/index.js`.

**Comment by trusktr** [comment]
Nevermind, `meteor add modules`. x}

**Comment by trusktr** [comment]
Will adding `modules` be required in Meteor 1.3?

**Comment by trusktr** [comment]
Nevermind the nevermind, after installing `modules`, files in `imports` are still being loaded. `app.js` in the Sources tab looks like

```js
//////////////////////////////////////////////////////////////////////////
//                                                                      //
// This is a generated file. You can view the original                  //
// source in your browser if your browser supports source maps.         //
// Source maps are supported by all recent versions of Chrome, Safari,  //
// and Firefox, and by Internet Explorer 11.                            //
//                                                                      //
//////////////////////////////////////////////////////////////////////////


(function () {

/* Imports */
var Foo = Package.mongo.Foo;
// a bunch of other things imported via `Package.*`

var require = meteorInstall({"app":{"client":{"imports":{"common":{"styles":{"reset.js":function(require,exports,module){

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                                                     //
// client/imports/common/styles/reset.js                                                                               //
//                                                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
                                                                                                                       //
                                                                                                                       // 1
/* global reset */                                                                                                     // 2
export default {                                                                                                       // 3

// and the rest of the file...
```

which causes

```
Uncaught SyntaxError: Unexpected token export
```

What am I missing?

**Comment by trusktr** [comment]
I was able to get Meteor 1.3 `modules` working in one project, but I'm not sure why it doesn't work in this one: https://github.com/trusktr/site/tree/8a151f953458ae8080902e879d6cf3df4dc7a5ee

**Comment by trusktr** [comment]
I'm going to close this one and start newer more concise issue.

**Comment by trusktr** [comment]
Re-opening as I think I found what might be causing this weird behavior I'm having: I took an app that was on Meteor 1.2-preview with a `.meteor/packages` file that look like this:

```conf
meteor-platform
rocket:module
kadira:flow-router
```

Then I updated it to 1.2.1, and then 1.3-modules-beta.2, which made it look like this (before adding `modules`):

```conf
rocket:module
kadira:flow-router
meteor-base
mobile-experience
mongo
blaze-html-templates
session
jquery
tracker
logging
reload
random
ejson
spacebars
check
```

There seems to be something wrong with that, because here's what `.meteor/packages` looks like (before adding `modules`) for a freshly created app that I updated directly to 1.3-modules-beta.2:

```conf
meteor-base             # Packages every Meteor app needs to have
mobile-experience       # Packages for a great mobile UX
mongo                   # The database Meteor supports right now
blaze-html-templates    # Compile .html files into Meteor Blaze views
session                 # Client-side reactive dictionary for your app
jquery                  # Helpful client-side library
tracker                 # Meteor's client-side reactive programming library

standard-minifiers      # JS/CSS minifiers run for production mode
es5-shim                # ECMAScript 5 compatibility for older browsers.
ecmascript              # Enable ECMAScript2015+ syntax in app code

autopublish             # Publish all data to the clients (for prototyping)
insecure                # Allow all DB writes from clients (for prototyping)
```

As you can see, that package list is different than the one where I upgraded from 1.2-preview.

**Comment by avital** [comment]
@trusktr This thread is very confusing. Can you summarize what your current actual problem is?

**Comment by trusktr** [comment]
Make a Meteor 1.2 app, then update to Meteor 1.3 beta.2 and add the `modules` package. It seems that `.meteor/packages` doesn't get properly updated by Meteor, and it won't work. The workaround is to make a new empty Meteor 1.3 beta.2 app, then replace `meteor-platform` with the list of packages that you see in a default Meteor 1.3 beta.2 app.

This problem might need to be fixed before the release of 1.3 stable or it will break people's app when they update to 1.3.

**Comment by trusktr** [comment]
Hmmm, this might also be related to #5885. I'm not sure.

**Comment by avital** [comment]
We're going to look at this as part of a more general issue: #5830.

(I'm surprised that you got that issue upgrading from 1.2. I think 1.2 already lists the specific packages you need rather than `meteor-platform`)

**Comment by trusktr** [comment]
This problem might be gone in beta 3. I updated to beta.3 from a 1.2 app the other day, and instead of Meteor modifying `.meteor/packages`, Meteor left it the same, leaving the `meteor-platform` entry intact. Adding `modules` worked this time. This is unlike the previous behavior I mentioned when upgrading from beta.2, which *does* modify `.meteor/packages`.
